### PR TITLE
Prevent wpool_pool from crashing on dead workers with best_worker strategy.

### LINKED ISSUE #94: wpool fails to handle temporarily dead workers
In particular, the following code:
```erlang
  case erlang:process_info(erlang:whereis(Worker), message_queue_len) of
    {message_queue_len, 0} -> Worker;
    {message_queue_len, L} ->
      min_message_queue(Checked + 1, next_wpool(Wpool), [{L, Worker} | Found])
  end.
```
in wpool_pool.erl fails if the worker is restarting, whether for valid or invalid reasons.

The dead workers aren't handled in any strategy, though; they're just assumed to be always alive. That's a shame since {stop, ...} tuples appear to be handled on the surface.

### PR DESCRIPTION
**Pull request by ElectronicRU**
Also got rid of the microoptimization around zero-length queue, it only serves to make code more confusing, as lists:min would select that zero anyway. As the code there got even more confusing, I chose to get rid of it.

Closes #94.

### CODE REVIEW

File: src/wpool_pool.erl [hunk 1]
```diff
@@ -405,10 +405,14 @@ min_message_queue(Size, #wpool{size = Size}, Found) ->
   Worker;
 min_message_queue(Checked, Wpool, Found) ->
   Worker = worker_name(Wpool#wpool.name, Wpool#wpool.next),
-  case erlang:process_info(erlang:whereis(Worker), message_queue_len) of
-    {message_queue_len, 0} -> Worker;
-    {message_queue_len, L} ->
-      min_message_queue(Checked + 1, next_wpool(Wpool), [{L, Worker} | Found])
+  min_message_queue(Checked + 1, next_wpool(Wpool), [{queue_length(whereis(Worker)), Worker} | Found]).
```
Comment (elvisci) [inline src/wpool_pool.erl]: Line 408 is too long:   min_message_queue(Checked + 1, next_wpool(Wpool), [{queue_length(whereis(Worker)), Worker} | Found])..

**Comment by elbrujohalcon** [comment]
Smart move, @ElectronicRU 👍 … I'll run the tests locally, check that the coverage remains at 100% and then merge it :)

File: src/wpool_pool.erl [hunk 2]
```diff
@@ -405,10 +405,15 @@ min_message_queue(Size, #wpool{size = Size}, Found) ->
   Worker;
 min_message_queue(Checked, Wpool, Found) ->
   Worker = worker_name(Wpool#wpool.name, Wpool#wpool.next),
-  case erlang:process_info(erlang:whereis(Worker), message_queue_len) of
-    {message_queue_len, 0} -> Worker;
-    {message_queue_len, L} ->
-      min_message_queue(Checked + 1, next_wpool(Wpool), [{L, Worker} | Found])
+  QueueLength = queue_length(whereis(Worker)),
+  min_message_queue(Checked + 1, next_wpool(Wpool), [{QueueLength, Worker} | Found]).
```
Comment (elvisci) [inline src/wpool_pool.erl]: Line 409 is too long:   min_message_queue(Checked + 1, next_wpool(Wpool), [{QueueLength, Worker} | Found])..

File: src/wpool_pool.erl [hunk 3]
```diff
@@ -405,10 +405,15 @@ min_message_queue(Size, #wpool{size = Size}, Found) ->
   Worker;
 min_message_queue(Checked, Wpool, Found) ->
   Worker = worker_name(Wpool#wpool.name, Wpool#wpool.next),
-  case erlang:process_info(erlang:whereis(Worker), message_queue_len) of
-    {message_queue_len, 0} -> Worker;
-    {message_queue_len, L} ->
-      min_message_queue(Checked + 1, next_wpool(Wpool), [{L, Worker} | Found])
+  QLength = queue_length(whereis(Worker)),
+  min_message_queue(Checked + 1, next_wpool(Wpool), [{QLength, Worker} | Found]).
```
Comment (elvisci) [inline src/wpool_pool.erl]: Line 409 is too long:   min_message_queue(Checked + 1, next_wpool(Wpool), [{QLength, Worker} | Found])..

Comment (ElectronicRU) [inline src/wpool_pool.erl]: God damn you, what is the line length configured on this thing.

File: src/wpool_pool.erl [hunk 4]
```diff
@@ -220,32 +220,36 @@ stats(Wpool, Sup) ->
...
+                [ message_queue_len
+                , memory
+                , current_function
+                , current_location
+                , dictionary
+                ]),
+            Time =
+              calendar:datetime_to_gregorian_seconds(calendar:universal_time()),
+            WS =
+              case {Function, proplists:get_value(wpool_task, Dictionary)} of
```
Comment (elvisci) [inline src/wpool_pool.erl]: The expression on line 240 and column 15 is nested beyond the maximum level of 3.

**Comment by ElectronicRU** [comment]
@elbrujohalcon I did as you said and also covered another potentially problematic case.

**Pull request merged**